lib/matplotlib/finance.py

```diff
@@ -1139,8 +1139,8 @@ def candlestick2_ohlc(ax, opens, highs, lows, closes, width=4,
     ax.autoscale_view()
 
     # add these last
-    ax.add_collection(barCollection)
     ax.add_collection(rangeCollection)
+    ax.add_collection(barCollection)
```

Comment:
OK, that would be a good thing to fix.  The way to do it is by setting the zorder kwarg of the barCollection to a value above that of the rangeCollection.  Where the barCollection is created, add a line right before the ending curly brace:
```python
zorder=rangeCollection.get_zorder() + 0.1,
```